index.ts

```diff
@@ -0,0 +1,190 @@
+import * as fs from "fs/promises";
+
+import chalk from "ansi-colors";
+import * as core from "@actions/core";
+import { ethers } from "ethers";
+
+process.on("unhandledRejection", (reason: any, _) => {
+    let error = `Unhandled Rejection occurred. ${reason.stack}`;
+    core.setFailed(error);
+});
+
+type ArtifactEntry = {
+    artifactPath: string;
+    sourcePath: string;
+    name: string;
+    address: string;
+    txHash?: string;
+};
+
+const greenCheck = chalk.green("✓");
+const redCross = chalk.red("×");
+
+(async function main() {
+    const registryPath = core.getInput("file");
+    const registry: ArtifactEntry[] = JSON.parse(
+        await fs.readFile(registryPath, { encoding: "utf8" })
+    );
+
+    let provider: ethers.providers.Provider;
+    const rpcUrl = core.getInput("rpc-url");
+    if (rpcUrl) {
+        provider = new ethers.providers.StaticJsonRpcProvider(rpcUrl);
+    } else {
+        // make it available to use the action without dedicated RPC endpoint
+        provider = new ethers.providers.InfuraProvider();
+    }
+
+    const promises = registry.map(async (desc: ArtifactEntry) => {
+        const isMatched = await verifyBytecode(desc, provider);
+        return {
+            isMatched,
+            desc,
+        };
+    });
+    const results = await Promise.allSettled(promises);
+
+    const sortedFulfilled = results
+        .filter((e) => e.status === "fulfilled")
+        .sort((a: any, b: any) => -a.value?.isMatched + b.value?.isMatched);
+
+    for (const result of sortedFulfilled) {
+        const value = (result as any).value as {
+            desc: ArtifactEntry;
+            isMatched: boolean;
+        };
+        const msgParts = value.isMatched ? [greenCheck] : [redCross];
+        msgParts.push(" ");
+        msgParts.push(value.desc.name);
+        msgParts.push(chalk.grey("@"));
+        msgParts.push(chalk.blue(value.desc.address));
+        const msg = msgParts.join("");
+
+        if (value.isMatched) {
+            core.info(msg);
+            continue;
+        }
+
+        core.setFailed(chalk.red(msg));
```

Comment:
<img width="635" alt="image" src="https://user-images.githubusercontent.com/31664571/203037706-5ba136f0-414e-43e1-b79b-e1d87400fb2b.png">
looks not so good

fail messages (seems to be) already red

Reply:
also failed check doesn't explain why it failed and what can (or should) I do with this, it is just the contract name with red cross

Reply:
There is no simple answer to why a verification process has failed and you should do research by yourself in any case.

Reply:
Redudant ANSI sequences removed.

Reply:
I think a message like 'Contract MyAwesomeContract.sol bytecode differs from deployed byte code' will be better
look at this from the developer who doesn't know what actions check their contract point of view

Reply:
Good point, I've updated the message.